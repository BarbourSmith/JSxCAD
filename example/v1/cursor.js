import '@jsxcad/api-v1-pdf';

import { Cursor } from '@jsxcad/api-v1-cursor';

Cursor.fromOrigin()
  .move(1)
  .rotateZ(90)
  .move(1)
  .rotateZ(90)
  .move(1)
  .rotateZ(90)
  .move(1)
  .rotateZ(90)
  .toShape()
  .Page()
  .view()
  .writePdf('cursor');
