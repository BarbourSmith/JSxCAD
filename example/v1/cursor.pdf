%PDF-1.5
1 0 obj << /Pages 2 0 R /Type /Catalog >> endobj
2 0 obj << /Count 1 /Kids [ 3 0 R ] /Type /Pages >> endobj
3 0 obj <<
  /Contents 4 0 R
  /MediaBox [ 0 0 595.275590176 841.889763249 ]
  /Parent 2 0 R
  /Type /Page
>>
endobj
4 0 obj << >>
stream
0.096000000 w
0.000000000 840.349076651 m
2.834645668 840.349076651 l
5.298897380 841.009370907 l
7.566613914 841.009370907 l
9.483254135 841.522933107 l
11.184041536 841.522933107 l
12.553070265 841.889763249 l
S
endstream
endobj
trailer << /Root 1 0 R /Size 4 >>
%%EOF