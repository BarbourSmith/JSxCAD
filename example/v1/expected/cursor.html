<html>
  <head>
    <title>JSxCAD Viewer</title>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0"
    />
  </head>
  <body>
    <script type="module">
      import { toDomElement } from 'https://gitcdn.link/cdn/jsxcad/JSxCAD/master/es6/jsxcad-ui-notebook.js';

      const notebook = [
        {
          view: {
            width: 512,
            height: 256,
            position: [100, -100, 100],
            geometry: {
              type: 'layout',
              layout: { size: [3, 3], margin: 1, title: '3 x 3 : ' },
              marks: [
                [0, 0, 0],
                [3, 3, 0],
              ],
              content: [
                {
                  type: 'item',
                  content: [
                    {
                      type: 'disjointAssembly',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'paths',
                              paths: [
                                [
                                  null,
                                  [-0.5, -0.5, 0],
                                  [0.5, -0.5, 0],
                                  [0.5, 0.5, 0],
                                  [-0.5, 0.5, 0],
                                  [-0.5, -0.5, 0],
                                ],
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                {
                  type: 'disjointAssembly',
                  content: [
                    {
                      type: 'disjointAssembly',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'paths',
                              tags: ['compose/non-positive', 'color/red'],
                              paths: [
                                [
                                  [1.5000000000000002, 1.5, 0],
                                  [-1.5, 1.5000000000000002, 0],
                                  [-1.5000000000000004, -1.4999999999999998, 0],
                                  [1.4999999999999998, -1.5000000000000004, 0],
                                ],
                              ],
                            },
                          ],
                        },
                        {
                          type: 'paths',
                          tags: ['compose/non-positive', 'color/red'],
                          paths: [
                            [
                              null,
                              [-1.4665178571428572, 1.734375, 0],
                              [-1.3928571428571428, 1.734375, 0],
                              [-1.4330357142857142, 1.6808035714285714, 0],
                              [-1.4129464285714286, 1.6808035714285714, 0],
                              [-1.3995535714285714, 1.6741071428571428, 0],
                              [-1.3928571428571428, 1.6674107142857142, 0],
                              [-1.3861607142857142, 1.6473214285714286, 0],
                              [-1.3861607142857142, 1.6339285714285714, 0],
                              [-1.3928571428571428, 1.6138392857142858, 0],
                              [-1.40625, 1.6004464285714286, 0],
                              [-1.4263392857142858, 1.59375, 0],
                              [-1.4464285714285714, 1.59375, 0],
                              [-1.4665178571428572, 1.6004464285714286, 0],
                              [-1.4732142857142858, 1.6071428571428572, 0],
                              [-1.4799107142857142, 1.6205357142857142, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-1.2388392857142858, 1.6875, 0],
                              [-1.1651785714285714, 1.59375, 0],
                            ],
                            [
                              null,
                              [-1.1651785714285714, 1.6875, 0],
                              [-1.2388392857142858, 1.59375, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-1.0044642857142858, 1.734375, 0],
                              [-0.9308035714285715, 1.734375, 0],
                              [-0.9709821428571429, 1.6808035714285714, 0],
                              [-0.9508928571428572, 1.6808035714285714, 0],
                              [-0.9375, 1.6741071428571428, 0],
                              [-0.9308035714285715, 1.6674107142857142, 0],
                              [-0.9241071428571429, 1.6473214285714286, 0],
                              [-0.9241071428571429, 1.6339285714285714, 0],
                              [-0.9308035714285715, 1.6138392857142858, 0],
                              [-0.9441964285714286, 1.6004464285714286, 0],
                              [-0.9642857142857143, 1.59375, 0],
                              [-0.984375, 1.59375, 0],
                              [-1.0044642857142858, 1.6004464285714286, 0],
                              [-1.0111607142857144, 1.6071428571428572, 0],
                              [-1.0178571428571428, 1.6205357142857142, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-0.7633928571428572, 1.6875, 0],
                              [-0.7700892857142858, 1.6808035714285714, 0],
                              [-0.7633928571428572, 1.6741071428571428, 0],
                              [-0.7566964285714286, 1.6808035714285714, 0],
                              [-0.7633928571428572, 1.6875, 0],
                            ],
                            [
                              null,
                              [-0.7633928571428572, 1.6071428571428572, 0],
                              [-0.7700892857142858, 1.6004464285714286, 0],
                              [-0.7633928571428572, 1.59375, 0],
                              [-0.7566964285714286, 1.6004464285714286, 0],
                              [-0.7633928571428572, 1.6071428571428572, 0],
                            ],
                            [null],
                            [null],
                          ],
                        },
                      ],
                    },
                    {
                      type: 'sketch',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'disjointAssembly',
                              content: [
                                {
                                  type: 'paths',
                                  tags: ['color/red'],
                                  paths: [
                                    [
                                      [1.5000000000000002, 1.5, 0],
                                      [-1.5, 1.5000000000000002, 0],
                                      [
                                        -1.5000000000000004,
                                        -1.4999999999999998,
                                        0,
                                      ],
                                      [
                                        1.4999999999999998,
                                        -1.5000000000000004,
                                        0,
                                      ],
                                    ],
                                  ],
                                },
                              ],
                            },
                            {
                              type: 'paths',
                              tags: ['color/red'],
                              paths: [
                                [
                                  null,
                                  [-1.4665178571428572, 1.734375, 0],
                                  [-1.3928571428571428, 1.734375, 0],
                                  [-1.4330357142857142, 1.6808035714285714, 0],
                                  [-1.4129464285714286, 1.6808035714285714, 0],
                                  [-1.3995535714285714, 1.6741071428571428, 0],
                                  [-1.3928571428571428, 1.6674107142857142, 0],
                                  [-1.3861607142857142, 1.6473214285714286, 0],
                                  [-1.3861607142857142, 1.6339285714285714, 0],
                                  [-1.3928571428571428, 1.6138392857142858, 0],
                                  [-1.40625, 1.6004464285714286, 0],
                                  [-1.4263392857142858, 1.59375, 0],
                                  [-1.4464285714285714, 1.59375, 0],
                                  [-1.4665178571428572, 1.6004464285714286, 0],
                                  [-1.4732142857142858, 1.6071428571428572, 0],
                                  [-1.4799107142857142, 1.6205357142857142, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-1.2388392857142858, 1.6875, 0],
                                  [-1.1651785714285714, 1.59375, 0],
                                ],
                                [
                                  null,
                                  [-1.1651785714285714, 1.6875, 0],
                                  [-1.2388392857142858, 1.59375, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-1.0044642857142858, 1.734375, 0],
                                  [-0.9308035714285715, 1.734375, 0],
                                  [-0.9709821428571429, 1.6808035714285714, 0],
                                  [-0.9508928571428572, 1.6808035714285714, 0],
                                  [-0.9375, 1.6741071428571428, 0],
                                  [-0.9308035714285715, 1.6674107142857142, 0],
                                  [-0.9241071428571429, 1.6473214285714286, 0],
                                  [-0.9241071428571429, 1.6339285714285714, 0],
                                  [-0.9308035714285715, 1.6138392857142858, 0],
                                  [-0.9441964285714286, 1.6004464285714286, 0],
                                  [-0.9642857142857143, 1.59375, 0],
                                  [-0.984375, 1.59375, 0],
                                  [-1.0044642857142858, 1.6004464285714286, 0],
                                  [-1.0111607142857144, 1.6071428571428572, 0],
                                  [-1.0178571428571428, 1.6205357142857142, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-0.7633928571428572, 1.6875, 0],
                                  [-0.7700892857142858, 1.6808035714285714, 0],
                                  [-0.7633928571428572, 1.6741071428571428, 0],
                                  [-0.7566964285714286, 1.6808035714285714, 0],
                                  [-0.7633928571428572, 1.6875, 0],
                                ],
                                [
                                  null,
                                  [-0.7633928571428572, 1.6071428571428572, 0],
                                  [-0.7700892857142858, 1.6004464285714286, 0],
                                  [-0.7633928571428572, 1.59375, 0],
                                  [-0.7566964285714286, 1.6004464285714286, 0],
                                  [-0.7633928571428572, 1.6071428571428572, 0],
                                ],
                                [null],
                                [null],
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          },
        },
      ];

      const run = async () => {
        const body = document.getElementsByTagName('body')[0];
        const notebookElement = await toDomElement(notebook);
        body.appendChild(notebookElement);
      };

      document.onreadystatechange = () => {
        if (document.readyState === 'complete') {
          run();
        }
      };
    </script>
  </body>
</html>
